docs/alb-ingress-rosa-hcp.md

```diff
@@ -0,0 +1,155 @@
+# Use AWS ALB as alternative ingress on ROSA HCP
+
+This doc aims at showing the effort needed to expose the OpenShift console via AWS ALB on a ROSA HCP cluster.
+The use case in mind is [HyperShift hosted clusters where the Ingress capability is disabled](https://github.com/openshift/enhancements/pull/1415).
+
+## Requirements
+
+- ROSA HCP OpenShift cluster.
+- [AWS Load Balancer Operator installed and its controller created](https://docs.openshift.com/rosa/networking/aws-load-balancer-operator.html).
+- User logged as a cluster admin.
+
+## Procedure
+
+### Create certificate in AWS Certificate Manager
+
+In order to configure an HTTPS listener on AWS ALB you need to have a certificate created in AWS Certificate Manager.
+You can import an existing certificate or request a new one. Make sure the certificate is created in the same region as your cluster.
+Note the certificate ARN, you will need it later.
+
+### Create NodePort services for the console
+
+The AWS Load Balancer Controller routes the traffic between cluster nodes.
+Therefore the services need to be exposed via a port on the node network:
+```bash
+cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: console-np
+  namespace: openshift-console
+spec:
+  ports:
+  - name: https
+    port: 443
+    protocol: TCP
+    targetPort: 8443
+  selector:
+    app: console
+    component: ui
+  type: NodePort
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: downloads-np
+  namespace: openshift-console
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: console
+    component: downloads
+  type: NodePort
+EOF
+```
+
+### Create Ingress resources for the services
+
+To provision ALBs create the following resources:
```

Comment:
> Wondering if this could be also done automatically by the console-operator if ALB setup is detected?

So far, we didn't see much automation around the certificate creation from the ALBO users. Probably because this requires quite some some effort to manage the cloud credentials. We can think about it as a followup enhancement. .